Microsoft.Toolkit.Uwp.Input.GazeInteraction/GazeTargetItem.cs

```diff
@@ -0,0 +1,239 @@
+                else if (peer.GetPattern(PatternInterface.SelectionItem) is ISelectionItemProvider)
+                {
+                    item = new SelectionItemPatternGazeTargetItem(element);
+                }
+                else if (peer.GetPattern(PatternInterface.ExpandCollapse) is IExpandCollapseProvider)
+                {
+                    item = new ExpandCollapsePatternGazeTargetItem(element);
+                }
+                else if (peer is ComboBoxItemAutomationPeer)
+                {
+                    item = new ComboBoxItemGazeTargetItem(element);
+                }
+                else
+                {
+                    item = GazePointer.Instance.NonInvokeGazeTargetItem;
+                }
+
+                element.SetValue(_gazeTargetItemProperty, item);
+            }
+
+            return item;
+        }
+
+        internal abstract void Invoke();
+
+        internal virtual bool IsInvokable
+        {
+            get { return true; }
+        }
+
+        internal void Reset(TimeSpan nextStateTime)
+        {
+            ElementState = PointerState.PreEnter;
+            DetailedTime = TimeSpan.Zero;
+            OverflowTime = TimeSpan.Zero;
+            NextStateTime = nextStateTime;
+            RepeatCount = 0;
+            MaxDwellRepeatCount = GazeInput.GetMaxDwellRepeatCount(TargetElement);
+        }
+
+        internal void GiveFeedback()
+        {
+            if (_nextStateTime != NextStateTime)
+            {
+                _prevStateTime = _nextStateTime;
+                _nextStateTime = NextStateTime;
+            }
+
+            if (ElementState != _notifiedPointerState)
+            {
+                switch (ElementState)
+                {
+                    case PointerState.Enter:
+                        RaiseProgressEvent(DwellProgressState.Fixating);
+                        break;
+
+                    case PointerState.Dwell:
+                    case PointerState.Fixation:
+                        RaiseProgressEvent(DwellProgressState.Progressing);
+                        break;
+
+                    case PointerState.Exit:
+                    case PointerState.PreEnter:
+                        RaiseProgressEvent(DwellProgressState.Idle);
+                        break;
+                }
+
+                _notifiedPointerState = ElementState;
+            }
+            else if (ElementState == PointerState.Dwell || ElementState == PointerState.Fixation)
+            {
+                if (RepeatCount <= MaxDwellRepeatCount)
+                {
+                    RaiseProgressEvent(DwellProgressState.Progressing);
+                }
+                else
+                {
+                    RaiseProgressEvent(DwellProgressState.Complete);
+                }
+            }
+        }
+
+        private void RaiseProgressEvent(DwellProgressState state)
+        {
+            // TODO: We should eliminate non-invokable controls before we arrive here!
```

Comment:
Is this why combobox isn't working as we filter it out here?

Reply:
I'm not sure. We need to track this on a different issue.